src/argilla_sdk/records/_dataset_records.py

```diff
@@ -382,11 +396,106 @@ def _normalize_batch_size(self, batch_size: int, records_length, max_value: int)
 
         return norm_batch_size
 
-    def __validate_vector_names(self, vector_names: Union[List[str], str]) -> None:
+    def _validate_vector_names(self, vector_names: Union[List[str], str]) -> None:
         if not isinstance(vector_names, list):
             vector_names = [vector_names]
         for vector_name in vector_names:
             if isinstance(vector_name, bool):
                 continue
             if vector_name not in self.__dataset.schema:
                 raise ValueError(f"Vector field {vector_name} not found in dataset schema.")
+
+    def _infer_record_from_mapping(
+        self,
+        data: dict,
+        mapping: Optional[Dict[str, str]] = None,
+        user_id: Optional[UUID] = None,
+    ) -> "Record":
+        """Converts a mapped record dictionary to a Record object for use by the add or update methods.
+        Args:
+            dataset: The dataset object to which the record belongs.
+            data: A dictionary representing the record.
+            mapping: A dictionary mapping source data keys to Argilla fields, questions, and ids.
+            user_id: The user id to associate with the record responses.
+        Returns:
+            A Record object.
+        """
+        fields: Dict[str, str] = {}
+        responses: List[Response] = []
+        record_id: Optional[str] = None
+        suggestion_values = defaultdict(dict)
+        vectors: List[Vector] = []
+        metadata: Dict[str, MetadataValue] = {}
+
+        schema = self.__dataset.schema
+
+        for attribute, value in data.items():
+            schema_item = schema.get(attribute)
+            attribute_type = None
+            sub_attribute = None
+
+            # Map source data keys using the mapping
+            if mapping and attribute in mapping:
+                attribute_mapping = mapping.get(attribute)
+                attribute_mapping = attribute_mapping.split(".")
+                attribute = attribute_mapping[0]
+                schema_item = schema.get(attribute)
+                if len(attribute_mapping) > 1:
+                    attribute_type = attribute_mapping[1]
+                if len(attribute_mapping) > 2:
+                    sub_attribute = attribute_mapping[2]
+            elif schema_item is mapping is None and attribute != "id":
```

Comment:
is this part of the elif correct `schema_item is mapping is None`?

Reply:
Yes. It is entered when an item in data is:
- not in the schema
- not in the mapping
- does not have the key "id"

in which case it is skipped